packages/remix-dev/__tests__/flat-routes-test.ts

```diff
@@ -604,4 +597,26 @@ describe("flatRoutes", () => {
       `Route segment cannot contain a slash: .lol?/what? (in route ($[$dollabills]).([.]lol)[/](what)/([$]).$)`
     );
   });
+
+  test("should error when using `*` in an escaped route segment", () => {
+    let files: string[] = ["routes/about.[*].tsx"];
+
+    expect(() =>
+      flatRoutesUniversal(
+        APP_DIR,
+        files.map((file) => path.join(APP_DIR, file))
+      )
+    ).toThrowError("Escaped route segment cannot contain `*`");
+  });
+
+  test("should error when using `:` in an escaped route segment", () => {
+    let files: string[] = ["routes/about.[:name].tsx"];
+
+    expect(() =>
+      flatRoutesUniversal(
+        APP_DIR,
+        files.map((file) => path.join(APP_DIR, file))
+      )
+    ).toThrowError("Escaped route segment cannot contain `:`");
+  });
```

Comment:
I think we want to error out whenever `:` or `*` appear _anywhere_ in the route, not just being escaped.

Since `:` and `*` are not part of Remix's special route characters like `$` or `()`, they should be treated like literals. But since there's not way to convey `:` and `*` as literals for RR (escaped or not), they are footguns that we should prevent users from running into.

Reply:
ah right right